cogeo_mosaic/backends/dynamodb.py

```diff
@@ -231,16 +270,37 @@ def get_assets(self, x: int, y: int, z: int) -> List[str]:
 
     def _fetch_dynamodb(self, quadkey: str) -> Dict:
         try:
-            return self.table.get_item(Key={"quadkey": quadkey}).get("Item", {})
+            return self.table.get_item(
+                Key={"mosaicId": self.mosaic_name, "quadkey": quadkey}
+            ).get("Item", {})
         except ClientError as e:
             status_code = e.response["ResponseMetadata"]["HTTPStatusCode"]
             exc = _HTTP_EXCEPTIONS.get(status_code, MosaicError)
             raise exc(e.response["Error"]["Message"]) from e
 
-    def _table_exist(self) -> bool:
+    def _table_exists(self) -> bool:
         """Check if the Table already exists."""
         try:
             _ = self.table.table_status
             return True
         except self.table.meta.client.exceptions.ResourceNotFoundException:
             return False
+
+    def _mosaic_exists(self) -> bool:
+        """Check if the mosaic already exists in the Table."""
+        item = self.table.get_item(
+            Key={"mosaicId": self.mosaic_name, "quadkey": self._metadata_quadkey}
+        ).get("Item", {})
+
+        return True if item else False
+
+    def clean(self):
```

Comment:
I think you should rename to something like `delete`. `clean` doesn't necessarily convey that it's a destructive action.